lambda/js/src/test/scala/feral/lambda/ExportedLambdaSuite.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda
+
+import cats.effect.IO
+import cats.effect.Ref
+import cats.effect.kernel.Resource
+import cats.syntax.all._
+import io.circe.scalajs._
+import munit.CatsEffectSuite
+
+import scala.scalajs.js
+import scala.scalajs.js.|
+
+class ExportedLambdaSuite extends CatsEffectSuite {
+  test("exported lambda") {
+    val context = DummyContext.asInstanceOf[facade.Context]
+
+    for {
+      allocationCounter <- IO.ref(0)
+      invokeCounter <- IO.ref(0)
+      lambda = new CountingIOLambda(allocationCounter, invokeCounter)
+
+      _ <- ('0' to 'z')
+        .map(_.toString)
+        .toList
+        .traverse(x =>
+          IO.fromPromise(IO(lambda.impl(x, context)))
+            .assertEquals(x.asJsAny.asInstanceOf[js.Any | Unit]))
+
+      _ <- allocationCounter.get.assertEquals(1)
+      _ <- invokeCounter.get.assertEquals(75)
+    } yield ()
+
+  }
+}
+
+class CountingIOLambda(allocationCounter: Ref[IO, Int], invokeCounter: Ref[IO, Int])
+    extends IOLambda[String, String] {
+
+  override def handler: Resource[IO, LambdaEnv[IO, String] => IO[Option[String]]] =
+    Resource
+      .eval(allocationCounter.getAndUpdate(_ + 1))
+      .as(_.event.map(Some(_)) <* invokeCounter.getAndUpdate(_ + 1))
+
+  // This would be exported with `@JSExportTopLevel("handler")`
+  def impl: HandlerFn = handlerFn
```

Comment:
Since we're not exporting it, what is this actually testing?

To verify the export works, our test should be able to call the exported method without knowing about the `CountingIOLambda` type.

Reply:
Yeah, I suppose this only tests that `impl` calls `handlerFn` which calls `handler`. Which verifies that the handler works when implemented by the class. I don't think you can explicitly call an export inside a unit test. If you have any ideas about how to approach I'd be happy to hear it

Reply:
I've now added a new sbt scripted test to replace this one. It's similar to the existing one, except it tests the exported handler

Reply:
> I don't think you can explicitly call an export inside a unit test. If you have any ideas about how to approach I'd be happy to hear it

For the record, I'm pretty sure you can :) conceptually an "exported" function is equivalent to a function defined in JavaScript land. So we can write a facade for it, and call it, without knowing that it is in fact defined in Scala!